<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-1-g15f4949  Build Version: 1.6.7594.29634
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="findchessbot">
  <link
    name="base_link">
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/base_link.STL" />
      </geometry>
    </collision>
  </link>
  
  <link
    name="link_1">
    <inertial>
      <origin
        xyz="-0.0431929949276883 -0.0167696838086873 0.215913559427298"
        rpy="0 0 0" />
      <mass
        value="4.57038820087514" />
      <inertia
        ixx="0.0122265464299465"
        ixy="4.21147542197394E-07"
        ixz="1.99180526350802E-05"
        iyy="0.0133372298042472"
        iyz="2.34384395342257E-06"
        izz="0.0085867023997394" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/link_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/link_1.STL" />
      </geometry>
    </collision>
  </link>


  <joint
    name="joint_1"
    type="revolute">
    <origin
      xyz="0 0 0.101"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="link_1" />
    <axis
      xyz="0 0 1" />
      <limit lower="-1.57" upper="1.57" effort="1.6" velocity="10" />
  </joint>

  <link
    name="link_2">
    <inertial>
      <origin
        xyz="0.169588884134961 0.000287079048425898 0.00493814032653461"
        rpy="0 0 0" />
      <mass
        value="1.68976202819323" />
      <inertia
        ixx="0.000578868076247387"
        ixy="-1.82793365282542E-06"
        ixz="3.45945381408847E-07"
        iyy="0.0094659655681031"
        iyz="6.38868613068616E-07"
        izz="0.00977868856241812" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/link_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/link_2.STL" />
      </geometry>
    </collision>
  </link>

  <joint
    name="joint_2"
    type="prismatic">
    <origin
      xyz="0.013249999999999 0 0.289"
      rpy="0 0 0" />
    <parent
      link="link_1" />
    <child
      link="link_2" />
    <axis
      xyz="0 0 1" />
      <limit lower="-0.1" upper="0" effort="1.6" velocity="10" />
  </joint>
  <link
    name="link_3">
    <inertial>
      <origin
        xyz="0.15471664673548 -5.19428621775786E-06 0.0034905551408036"
        rpy="0 0 0" />
      <mass
        value="0.802329350003151" />
      <inertia
        ixx="0.000293485369809973"
        ixy="-6.26782699018864E-09"
        ixz="1.27148249238739E-07"
        iyy="0.00409323307841445"
        iyz="-4.78551932884101E-08"
        izz="0.00412215164530587" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/link_3.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/link_3.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_3"
    type="revolute">
    <origin
      xyz="0.37 0 0.0649999999999992"
      rpy="0 0 0" />
    <parent
      link="link_2" />
    <child
      link="link_3" />
    <axis
      xyz="0 0 1" />
      <limit lower="-3.14" upper="3.14" effort="0" velocity="10" />
  </joint>
  <link
    name="link_4">
    <inertial>
      <origin
        xyz="7.69925345344147E-05 2.22729360906393E-05 -0.0787289861832771"
        rpy="0 0 0" />
      <mass
        value="0.427295816857813" />
      <inertia
        ixx="0.000121089035690954"
        ixy="9.95625908929604E-08"
        ixz="-3.89978466575397E-07"
        iyy="0.000119966189995005"
        iyz="-7.03773299579463E-09"
        izz="0.000127824456429408" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/link_4.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/link_4.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="joint_4"
    type="revolute">
    <origin
      xyz="0.315000000000002 0 0.000500000000001111"
      rpy="0 0 0" />
    <parent
      link="link_3" />
    <child
      link="link_4" />
    <axis
      xyz="0 0 1" />
      <limit lower="-3.14" upper="3.14" effort="0" velocity="10" />
  </joint>
  <link
    name="chess_board">
    <inertial>
      <origin
        xyz="1.1102E-16 -5.5511E-17 -0.01591"
        rpy="0 0 0" />
      <mass
        value="2.2955" />
      <inertia
        ixx="0.017747"
        ixy="2.9956E-19"
        ixz="-3.1844E-19"
        iyy="0.021703"
        iyz="2.6072E-18"
        izz="0.039234" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/chessboard.STL" />
      </geometry>
      <material
        name="">
        <color
        rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/chessboard.STL" />
      </geometry>
    </collision>
  </link>


  <joint
    name="joint_chess"
    type="revolute">
    <origin
      xyz="0.42744 -0.00059371 0.099999"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="chess_board" />
    <axis
      xyz="0 0 1" />
      <limit lower="-6.28" upper="6.28" effort="0" velocity="10" />
  </joint>

  
  <link
    name="chess_board_black">
    <visual>
      <origin
        xyz="0 0 0.001"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/chessboard_black.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0 0 0 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://findchessbot/chessboard_black.STL" />
      </geometry>
    </collision>
  </link>

  <joint
  name="joint_chess_black"
  type="fixed">
  <parent
    link="chess_board" />
  <child
    link="chess_board_black" />
</joint>

</robot>